components/brave_ads/core/internal/common/locale/language_code_android.cc

```diff
@@ -0,0 +1,24 @@
+/* Copyright (c) 2025 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#include "brave/components/brave_ads/core/internal/common/locale/language_code.h"
+
+#include <optional>
+#include <string>
+
+#include "base/android/locale_utils.h"
+
+namespace brave_ads {
+
+std::optional<std::string> MaybeGetLanguageCodeString() {
+  std::string locale = base::android::GetDefaultLocaleString();
+  const size_t pos = locale.find_first_of("-_");
+  if (pos != std::string::npos) {
```

Comment:
we can just return `locale.substr(0, pos);` because `locale.substr(0, npos)` returns whole string.

Reply:
Fixed